.github/workflows/check-packages.yml

```diff
@@ -3,6 +3,35 @@ on:
   pull_request:
 
 jobs:
+  dependency-deduplication:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14.18.1
+
+      - uses: tj-actions/changed-files@v32
+        id: changed-files-specific
+        with:
+          files: |
+            yarn.lock
+
+      - name: Check package duplicates
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
+        run: |
+          echo "yarn.lock changed! Verifying package deduplication ..."
```

Comment:
Should we also add this as a pre commit hook once all dupes are removed?

Reply:
probably within pre-push. commits should be blazing fast (good DX). this check is definitely not fast.